instant-xml/src/impls.rs

```diff
@@ -25,3 +29,143 @@ impl<'xml> FromXml<'xml> for bool {
+            Some(FieldAttribute::Attribute) => {
+                serializer.add_attribute_value(&self.0)?;
+            }
+            _ => {
+                serializer.add_open_tag(&field_context)?;
+                write!(serializer.output, "{}", self.0)?;
+                serializer.add_close_tag(field_context)?;
+            }
+        }
+        Ok(())
+    }
+}
+
+macro_rules! to_xml_for_number {
+    ($typ:ty) => {
+        impl ToXml for $typ {
+            fn serialize<W>(&self, serializer: &mut Serializer<W>) -> Result<(), Error>
+            where
+                W: fmt::Write,
+            {
+                DisplayToXml(self).serialize(serializer)
+            }
+        }
+    };
+}
+
+to_xml_for_number!(i8);
+to_xml_for_number!(i16);
+to_xml_for_number!(i32);
+to_xml_for_number!(i64);
+to_xml_for_number!(isize);
+to_xml_for_number!(u8);
+to_xml_for_number!(u16);
+to_xml_for_number!(u32);
+to_xml_for_number!(u64);
+to_xml_for_number!(usize);
+to_xml_for_number!(f32);
+to_xml_for_number!(f64);
+
+impl ToXml for bool {
+    fn serialize<W>(&self, serializer: &mut Serializer<W>) -> Result<(), Error>
+    where
+        W: fmt::Write,
+    {
+        let value = match self {
+            true => "true",
+            false => "false",
+        };
+
+        DisplayToXml(&value).serialize(serializer)
+    }
+}
+
+impl ToXml for String {
+    fn serialize<W>(&self, serializer: &mut Serializer<W>) -> Result<(), Error>
+    where
+        W: fmt::Write,
+    {
+        DisplayToXml(&escape(self)?).serialize(serializer)
+    }
+}
+
+impl ToXml for char {
+    fn serialize<W>(&self, serializer: &mut Serializer<W>) -> Result<(), Error>
+    where
+        W: fmt::Write,
+    {
+        let mut tmp = [0u8; 4];
+        DisplayToXml(&escape(&*self.encode_utf8(&mut tmp))?).serialize(serializer)
+    }
+}
+
+impl ToXml for &str {
+    fn serialize<W>(&self, serializer: &mut Serializer<W>) -> Result<(), Error>
+    where
+        W: fmt::Write,
+    {
+        DisplayToXml(&escape(self)?).serialize(serializer)
+    }
+}
+
+impl ToXml for Cow<'_, str> {
+    fn serialize<W>(&self, serializer: &mut Serializer<W>) -> Result<(), Error>
+    where
+        W: fmt::Write,
+    {
+        DisplayToXml(&escape(self)?).serialize(serializer)
+    }
+}
+
+fn escape(input: &str) -> Result<Cow<'_, str>, Error> {
+    let mut result = String::with_capacity(input.len());
+    let mut last_end = 0;
+    for (start, c) in input.chars().enumerate() {
+        let to = match c {
+            '&' => "&amp;",
+            '"' => "&quot;",
+            '<' => "&lt;",
+            '>' => "&gt;",
+            '\'' => "&apos;",
+            _ => continue,
+        };
+        match input.get(last_end..start) {
+            Some(v) => result.push_str(v),
+            None => return Err(Error::Other("Out of bounds".to_string())),
+        }
+
+        result.push_str(to);
+        last_end = start + 1;
+    }
+
+    if result.is_empty() {
+        Ok(Cow::Borrowed(input))
```

Comment:
Let's just `return` here, so we don't need the indented else block.

Reply:
done